File: homework_0/src/java/main/SmallestTwo.java
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class SmallestTwo {
    public static void main(String[] args) {
        Set<Integer> numbers = new HashSet<>();
        try (Scanner scanner = new Scanner(System.in)){
        //reading int the input
        while (scanner.hasNextInt()) {
            numbers.add(scanner.nextInt());
            }
        }

        //have a minimum variable and second minimum vairable
        int min = Integer.MAX_VALUE;
        int second_min = Integer.MAX_VALUE;

        for (int number : numbers) {
            if (number < min) {
                second_min = min;
                min = number;
            }
            else if (number < second_min) second_min = number;
        }

        //proitning out the min and second min
        System.out.println(min);
        System.out.println(second_min);

    }
        
    }

File: homework_0/smallesttwo.py
import sys

def main():
    #DOing a set so it doesn't have repeat
    numbers = set()

    # Read all ints from standard input
    for token in sys.stdin.read().split():
        numbers.add(int(token))

    # have a minimum variable and second minimum vairable
    min_val = float('inf')
    second_min = float('inf')

    for num in numbers:
        if num < min_val:
            second_min = min_val
            min_val = num
        elif num < second_min:
            second_min = num

    # proitning out the min and second min
    print(min_val)
    print(second_min)

if __name__ == "__main__":
    main()
